pyanaconda/localization.py

```diff
@@ -189,20 +189,25 @@ def find_best_locale_match(locale, langcodes):
     def get_match_score(locale, langcode):
         score = 0
 
-        locale_parts = parse_langcode(locale)
-        langcode_parts = parse_langcode(langcode)
-        if not locale_parts or not langcode_parts:
+        if not is_valid_langcode(locale) or not is_valid_langcode(langcode):
             return score
 
+        locale_parsed = langtable.parse_locale(locale)
+        langcode_parsed = langtable.parse_locale(langcode)
+
         for part, part_score in SCORE_MAP.items():
-            if locale_parts[part] and langcode_parts[part]:
-                if locale_parts[part] == langcode_parts[part]:
+
+            locale_value = locale_parsed.__getattribute__(part)
+            langcode_value = langcode_parsed.__getattribute__(part)
```

Comment:
The same here also I'm not sure if this will work. If I'm reading the documentation correctly then if the part is missing you will get `AttributeError` exception.

Reply:
The attributes are now always present, see https://github.com/mike-fabian/langtable/blob/master/langtable/langtable.py#L1100